### PR TITLE
feat: Add golangci-lint linters - imports

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Add more linting checks to project with `golangci-lint`. This PR fixes issues related to imports in the code.

`.golangci.yaml` is similar to the one proposed at https://github.com/vmware-tanzu/velero/pull/7194

## Test

Run `golangci-lint` with new `make lint` and `make lint-fix` commands.

## TODO

All go files changes where related to imports errors (with the help of `make lint-fix` command). There are still a lot of issues to be resolved in the code, but they will be fixed by categories in follow up PRs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -530,3 +524,17 @@ test-e2e-cleanup: login-required
 	$(OC_CLI) delete restore -n $(OADP_TEST_NAMESPACE) --all --wait=false
 	for restore_name in $(shell $(OC_CLI) get restore -n $(OADP_TEST_NAMESPACE) -o name);do $(OC_CLI) patch "$$restore_name" -n $(OADP_TEST_NAMESPACE) -p '{"metadata":{"finalizers":null}}' --type=merge;done
 	rm -rf $(SETTINGS_TMP)
+
+GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
+
+.PHONY: golangci-lint
+golangci-lint: ## Download golangci-lint locally if necessary.
+	$(call go-install-tool,$(GOLANGCI_LINT),github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2)
```
Comment (mateusoliveira43) [inline Makefile:532]: latest version that works with go `1.20`

**Comment by mateusoliveira43** [comment]
only error missing related for imports is dot-imports. Can I fix them or someone likes that?

also checks the current errors, if you want to add any more fix to the PR. Otherwise, will comment those, like I did with the other (which will be fixed in future follow up PRs)

**Review by shawn-hurley — COMMENTED** [review]
LGTM, I think this is an excellent start to standardizing imports!

**Comment by weshayutin** [comment]
@mateusoliveira43 thanks for pushing this forward and taking small steps ++

**Review by shubham-pampattiwar — APPROVED** [review]
changes look sane to me, VISIACK

**Comment by weshayutin** [comment]
```
whayutin@thinkdoe:~/OPENSHIFT/git/OADP/oadp-operator$ make lint
/home/whayutin/OPENSHIFT/git/OADP/oadp-operator/bin/golangci-lint run
controllers/registry.go:12: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/openshift/oadp-operator) (gci)
	"k8s.io/apimachinery/pkg/labels"
controllers/registry.go:17: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/openshift/oadp-operator) (gci)
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
make: *** [Makefile:542: lint] Error 1
whayutin@thinkdoe:~/OPENSHIFT/git/OADP/oadp-operator$ git status
On branch feat/enforce-linter-rules
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   controllers/registry.go

no changes added to commit (use "git add" and/or "git commit -a")
whayutin@thinkdoe:~/OPENSHIFT/git/OADP/oadp-operator$ make lint-fix
/home/whayutin/OPENSHIFT/git/OADP/oadp-operator/bin/golangci-lint run --fix
whayutin@thinkdoe:~/OPENSHIFT/git/OADP/oadp-operator$ git status
On branch feat/enforce-linter-rules
nothing to commit, working tree clean
whayutin@thinkdoe:~/OPENSHIFT/git/OADP/oadp-operator$ make lint
/home/whayutin/OPENSHIFT/git/OADP/oadp-operator/bin/golangci-lint run
```
Looks good to me, however to be perfectly honest, I'm not sure where the lint logs are in prow.  Looked in unit and didn't see it.

**Comment by mateusoliveira43** [comment]
@weshayutin in `unit-test` (the line `/go/src/github.com/openshift/oadp-operator/bin/golangci-lint run`, it would print more things, if there were errors)

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
will open cherry-pick PR once there are no oadp-1.3 PRs open

**Pull request merged**